eng/templates/stages/release.yml

```diff
@@ -0,0 +1,297 @@
+        query="{ repository(owner: \"dotnet\", name: \"$repo\") { refs(refPrefix: \"refs/tags/\", last: 100, query: \"$tag\") { nodes { name }}}}"
+        tags=$(gh api graphql -f query="$query" --template '{{.data.repository.refs.nodes}}')
+
+        if echo "$tags" | grep -q "map\[name:$tag\]"; then
+          echo "##vso[task.setvariable variable=Tag;isOutput=true]$tag"
+          echo "Tag $tag exists"
+        else
+          echo "##vso[task.logissue type=error]Tag $tag does not exist in dotnet/$repo"
+          exit 1
+        fi
+      displayName: Validate release tag
+      name: TagValidation
+      env:
+        GH_TOKEN: $(BotAccount-dotnet-sb-bot-pat)
+
+  - job: CreateReleaseAnnouncementJob
+    displayName: Create Release Announcement
+    condition: and(succeeded(), eq('${{ parameters.createReleaseAnnouncement }}', true))
+    dependsOn: ValidateReleaseTag
+    variables:
+    - name: Tag
+      value: $[ dependencies.ValidateReleaseTag.outputs['TagValidation.Tag'] ]
+    steps:
+    - script: |
+        set -euxo pipefail
+
+        query='query { repository(owner: "${{ variables.announcementOrg }}", name: "${{ variables.announcementRepo }}") { id } }'
+        echo "${query}"
+        repo_id=$( gh api graphql -f query="$query" --template '{{.data.repository.id}}' )
+        echo ${{ variables.announcementOrg }}/${{ variables.announcementRepo }} repo ID is ${repo_id}
+
+        query='query { repository(name: "${{ variables.announcementRepo }}", owner: "${{ variables.announcementOrg }}") { discussionCategories(first: 10) { edges { node { id, name } } } } }'
+        echo "${query}"
+        category_id=$( gh api graphql -f query="$query" --template '{{range .data.repository.discussionCategories.edges}}{{if eq .node.name "Announcements"}}{{.node.id}}{{end}}{{end}}' )
+        echo Discussion Category ID is ${category_id}
+
+        echo "##vso[task.setvariable variable=RepoId;]${repo_id}"
+        echo "##vso[task.setvariable variable=DiscussionCategoryId;]${category_id}"
+      displayName: Get category IDs
+      env:
+        GH_TOKEN: $(BotAccount-dotnet-sb-bot-pat)
+
+    - script: |
+        set -euxo pipefail
+        echo "Repo ID is $(RepoId)"
+        echo "Discussion Category ID is $(DiscussionCategoryId)"
+
+        # Set environment variables that go in the announcement template
+        export TAG=$(Tag)
+        export RUNTIME_VERSION="$(runtimeVersion)"
+        export RELEASE_CHANNEL="$(releaseChannel)"
+        export SDK_VERSION="$(sdkVersion)"
+        export RELEASE_NOTES_URL="https://github.com/dotnet/core/blob/main/release-notes/$RELEASE_CHANNEL/$RUNTIME_VERSION/$SDK_VERSION.md"
+        export RELEASE_DATE=$(date +"%B %Y") # e.g. "March 2022"
+
+        if [[ "$(releaseChannel)" == '6.0' || "$(releaseChannel)" == '7.0' ]]; then
+          export TAG_URL="https://github.com/dotnet/installer/releases/tag/$TAG"
+        else
+          export TAG_URL="https://github.com/dotnet/dotnet/releases/tag/$TAG"
```

Comment:
@MichaelSimons I'd need some clarification on how we will handle the final instructions in the announcement.

- We could send people directly to a commit URL in the VMR. We have this SHA available when this script is running.
- We could send them to the release (we need to create the release?) and we need to determine the content of the GH release.
- We should also still include instructions on how to set the version when building the SDK from the VMR. I was looking in the SB README but didn't find a way to specify this. We could include this piece of information in the release/announcement.